Doc/library/asyncio-stream.rst

```diff
@@ -367,13 +366,167 @@ Stream
 
    Represents a Stream object that provides APIs to read and write data
    to the IO stream . It includes the API provided by :class:`StreamReader`
-   and :class:`StreamWriter`.
+   and :class:`StreamWriter`. It can also be used as :term:`asynchronous iterator`
+   where :meth:`readline` is used. It raises :exc:`StopAsyncIteration` when
+   :meth:`readline` returns empty data.
 
    Do not instantiate *Stream* objects directly; use API like :func:`connect`
    and :class:`StreamServer` instead.
 
    .. versionadded:: 3.8
 
+   .. attribute:: mode
+
+      Returns the mode of the stream which is a :class:`StreamMode` value. It could
+      be one of the below :
+
+      * :attr:`StreamMode.READ` - Connection can receive data.
+      * :attr:`StreamMode.WRITE` - Connection can send data.
+      * :attr:`StreamMode.READWRITE` - Connection can send and receive data.
+
+   .. coroutinemethod:: abort()
+
+      Aborts the :ref:`Transport <asyncio-transport>` instance used by the Stream.
```

Comment:
This is a relatively low-level description of what's happening here.  I'd use a more high-level description, such as "Aborts the connection immediately, without waiting for the send buffer to drain." (please double check that :))

Reply:
I checked the method and it goes to call the abort method of the transport and waits for the close. I think it doesn't wait for data to be drained.

Reply:
Yes, `transport.abort()` clears pending write buffer if any,